include/swift/AST/PrintOptions.h

```diff
@@ -294,6 +275,9 @@ struct PrintOptions {
   /// Whether to print the extensions from conforming protocols.
   bool PrintExtensionFromConformingProtocols = false;
 
+  std::shared_ptr<ShouldPrintChecker> CurrentPrintabilityChecker =
```

Comment:
Small note: it seems a little funny to make this a `shared_ptr` rather than just an optional non-owning pointer.